src/charts/axis.js

```diff
@@ -196,6 +197,19 @@ export default function Axis (_container) {
     }
   }
 
+  function truncateTickText (axisSelection) {
+    if (!config.maxLabelCharCount) {
+      return 
+    }
+    axisSelection.selectAll(".tick text")
+      .each(function (d) {
+        const text = this.textContent
+        if (text.length > config.maxLabelCharCount) {
+          this.textContent = `${text.slice(0, config.maxLabelCharCount)}…`
```

Comment:
Should probably add `this.attr("title", text)` here so that the labels will still show a tooltip of the original text on hover.

Reply:
It's a good idea. I tried it, but it's more tricky than it looks. There's no title attribute in svg, but I added it as a title child node. (Also, technically, there's no requirement for the browser to use the title attribute or node as a tooltip.) But the real problem is that the domain editor is grabbing the events, so the hover on labels doesn't trigger the title to show-up. I will try a more complicated event bubbling or retriggering, but it's not as simple as setting a title attribute.

Reply:
@biovisualize there's not a `title` attribute for text elements but there is a `title` element in SVG:
https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title

But we'd have to append the title to the `g.tick` element not the `text` element.

Reply:
It works in multiple ways. One is like this, a title element as a child of the text element:
```
axisSelection.selectAll(".tick text")
      .each(function (d) {
        const text = this.textContent
        if (text.length > config.maxLabelCharCount) {
          this.textContent = `${text.slice(0, config.maxLabelCharCount)}…`
          console.dir(this)
          const title = d3.select(this)
            .selectAll("title")
            .data([text])
          title.enter().append("title")
            .merge(title)
            .text(d => d)  
          title.exit().remove()
        }
      })
```

Reply:
Indeed, I see that 
>Generally <title> element should be the first child element of its parent. Note that those implementations that do use <title> to display a tooltip often will only do so if the <title> is indeed the first child element of its parent.

But anyway, it works, I can easily polish it, but my problem is with the event handling. Nothing that can't be solved, but I would prefer to have a design decision than a technical solution for that https://github.com/mapd/mapd-immerse/issues/4363#issuecomment-393842261